docs/src/IconButton.doc.js

```diff
@@ -12,202 +12,357 @@ const card = c => cards.push(c);
+          `- 'link': Use for buttons to act like links. The button is rendered as an '<a>'.`,
+          `Required with link-role buttons.`,
+        ],
+        href: 'roles',
       },
       {
         name: 'selected',
         type: 'boolean',
+        required: false,
         defaultValue: false,
-        href: 'selectedCombinations',
+        description: [
+          'Control the "selected" state of a button component to toggle binary states.',
+          'Optional with button-role + button-type buttons.',
+        ],
+        href: 'selected',
       },
       {
         name: 'size',
         type: `"xs" | "sm" | "md" | "lg" | "xl"`,
-        description: `xs: 24px, sm: 32px, md: 40px, lg: 48px, xl: 56px`,
+        required: false,
         defaultValue: 'md',
-        href: 'sizeCombinations',
+        description:
+          'Display an IconButton in different sizes. Size changes the component padding modifying its fixed height: xs (24px), sm (32px), md (40px), lg (48px), xl (56px).',
+        href: 'size',
+      },
+      {
+        name: 'target',
+        type: `null | 'self' | 'blank'`,
+        required: false,
+        defaultValue: 'null',
+        description: [
+          'Define the frame or window to open the anchor defined on `href`:',
+          `- 'null' opens the anchor in the same window.`,
+          `- 'blank' opens the anchor in a new window.`,
+          `- 'self' opens an anchor in the same frame.`,
+          'Optional with link-role buttons.',
+        ],
+        href: 'roles',
       },
     ]}
   />
 );
 
 card(
   <Example
-    name="Example"
+    name="Basic IconButton"
+    id="basic"
     defaultCode={`
 <IconButton
-  accessibilityLabel="Love"
-  bgColor="white"
-  icon="heart"
-  iconColor="red"
-  onClick={() => { console.log('❤️')}}
+  accessibilityLabel="Open edit modal to edit Board"
+  bgColor="gray"
+  icon="edit"
 />
 `}
   />
 );
 
 card(
   <Example
-    id="accessibility-popup"
-    description={`
-      We want to make sure every button on the page is unique when being read by screenreader.
-      - \`accessibilityHaspopup\` specifies that the button has associated content (i.e. Flyout).
-      - \`accessibilityLabel\` updates the spoken text.
-  `}
-    name="Example: Accessibility (Popup)"
+    name="Roles"
+    id="roles"
     defaultCode={`
-function A11yExPopup() {
-  const [isOpen, setOpen] = React.useState(false);
-  const anchorRef = React.useRef(null);
-
+function Example() {
+  const [disabled, setDisabled] = React.useState(false);
   return (
-    <>
-      <Box display="inlineBlock" ref={anchorRef}>
-        <IconButton
-          accessibilityLabel="see more"
-          accessibilityHaspopup
-          icon="ellipsis"
-          iconColor="darkGray"
-          onClick={() => setOpen(!isOpen)}
+    <Stack gap={3}>
+      <Row gap={1}>
+        <Tooltip text="Default IconButton">
```

Comment:
@AlbertCarreras instead of these `<Tooltip />` - could we show a label next to it instead? (Make it more accessible + clear to our users)

Reply:
Labels only work with: SelectList, Switch, TextField or TextArea. I could add Text. I'll keep the pattern until we agree on the templates. I used this pattern in Button and TapArea as well.